training/model_management/src/azureml/model/mgmt/processors/transformers/whisper/predict.py

```diff
@@ -163,18 +147,23 @@ def audio_processor(audio_input: str, sampling_rate: int = 16000) -> np.array:
     with TemporaryDirectory() as temp_dir:
         audio_file_path = os.path.join(temp_dir, "audio_file.m4a")
 
-        # check if input string is a URI to an audio file
         if is_valid_url(audio_input):
-            with open(audio_file_path, "wb") as f:
-                f.write(requests.get(audio_input).content)
-            return audio_input_to_nparray(audio_file_path, sampling_rate)
-
-        try:
-            with open(audio_file_path, "wb") as audio:
-                audio.write(base64.b64decode(audio_input))
-        except Exception:
-            raise ValueError("Invalid audio format")
-        return audio_input_to_nparray(audio_file_path, sampling_rate)
+            logging.info("Recieved Remote Audio URL as input.")
+            try:
+                audio_file = wget.download(audio_input, out=audio_file_path)
```

Comment:
can we please not use wget, as that will introduce new dependency for model-mgmt env?

Reply:
using requests is causing ffmpeg conversion to failw with corrupt file error

Reply:
Could you add wget to the list of dependencies for model-mgmt env. writing response.content to a file is very vulnerable.

Reply:
wget internally make use of ulib.urlretreive. Can we do something like below?

```
import urllib.request

urllib.request.urlretrieve("[https://cdn-media.huggingface.co/speech_samples/sample1.flac"](https://cdn-media.huggingface.co/speech_samples/sample1.flac%22), audio_file_path)
```

Reply:
Moving ahead with e2e tested scenario for whisper. Added wget to training/model_management/environments/model_managements/context/requirements.txt